pyomo/core/base/component.py

```diff
@@ -366,6 +366,14 @@ def reconstruct(self, data=None):
         self._constructed = False
         self.construct(data=data)
 
+    def is_named_expression_type(self):
```

Comment:
Why do *components* need to duck type the expression API?

Reply:
I confirmed that this method isn't needed here.